.eslintrc.js

```diff
@@ -37,7 +37,7 @@ const addReactCompilerRule = (packagesNames, isEnabled) =>
   !isEnabled
     ? []
     : packagesNames.map((packageName) => ({
-        files: [`packages/${packageName}/src/**/*{.ts,.tsx,.js}`],
+        files: [`packages/${packageName}/src/**/*.?(c|m)[jt]s?(x)`],
```

Comment:
`.jsx` shouldn't be allowed.

With this regex, it seems hard to search in the codebase for, e.g. `.tsx`, I suspect that listing all the permutations would be clearer.

Reply:
I don't see why you would need to search for that though. As you will have to sweep over all configuration files when doing a change that would require updating that anyways 😅 

Replacing it to list all the permutations is quite unnecessary in my view: `{.ts,.tsx,.js,.cjs,.cjsx,.mjs,.mjsx,.cts,.ctsx,.mts,.mtsx}`

We could cleanup some unlikely exts like `{.ts,.tsx,.js,.cjs,.mjs,.mts}` which would be simpler, but less complete.

Reply:
> but less complete.

Could it actually be great? I assume we will never need those other extensions, so if someone create them, he might be more likely to realize those are wrong?

Reply:
> Could it actually be great? I assume we will never need those other extensions, so if someone create them, he might be more likely to realize those are wrong?

Quite the contrary, we sometimes need those extensions:

* When running code natively on node, we need a way to signal which module system is being used
* We need .cjs in edge-cases for upcoming ESM support in places where we want to bridge between ESM and CJS, e.g. to deal with importing `next/document` in ESM.
* Tooling like `tsx` and `vitest` rely on the extension to determine whether JSX needs to be transformed or not. This is currently giving problems as we have a bunch of jsx in `.js` files.

Reply:
> They likely won't realize anything

If they can't run the tests they try to write in dev mode, I think they will notice that something is off.

For eslint specifically, yeah, agree, the ideal is to lint the file and have an eslint rule that error because the extension is wrong.

Reply:
> If a developer can't run the tests he/she try to write in dev mode, I think he/she will notice that something is off.

In any case, why have them waste time reverse engineering the tests to find out they're using the wrong extension if the eslint plugin can tell them in context in the editor? Not that there would be anything wrong with using .jsx instead of .js for JSX files, contrary, it would make things easier.

Reply:
> if the eslint plugin can tell them in context in the editor?

👍 

> Not that there would be anything wrong with using .jsx instead of .js for JSX files

I think that the value of only having `.js` is about not having to think about extensions, not having to rename a file after refactoring its content, and moving the code around. It's friction to change.

Reply:
But we can't get around this friction in typescript. Finishing the typescript migration means imposing this friction across the codebase anyway.

Reply:
> If a developer can't run the tests he/she try to write in dev mode, I think he/she will notice that something is off.

🙂 Coincidentally, I'm just running into a problem that illustrates why this is harmful. Just trying to figure out why I have a failing test locally in vitest that doesn't seem to break in mocha. Yet, the following produces a failing test:

```
pnpm --filter @mui/icons-material test -- -g "should produce the expected output"
```

But when you try running the global command:

```
pnpm test -- -g "should produce the expected output"  
```

Whoops, no test is running. What happened? Somewhere along the way we started running the build script natively in node. The file was renamed to .mjs. The test runner only looks for .js, .ts, and .tsx. The author didn't see any tests break after running them and assumed they were good. They didn't realize the test stopped running altogether.